packages/webpack-plugin/src/compiler.ts

```diff
@@ -64,9 +64,15 @@ function compileVanillaSource(
       loader._compiler.webpack && loader._compiler.webpack.version,
     );
     const compat = createCompat(isWebpack5);
-    // Child compiler will compile vanilla-extract files to be evaled during compilation
-    const outputOptions = { filename: loader.resourcePath };
 
+    // Escape webpack template strings in output files so they don't get replaced
+    const outputOptions = {
+      filename: loader.resourcePath
+        .replaceAll('[', '[\\')
+        .replaceAll(']', '\\]'),
```

Comment:
Related to the other comment:

```suggestion
      filename: loader.resourcePath.replaceAll(/(\[|\])/g, '\\$1'),
```